pkg/asset/installconfig/ovirt/credentials.go

```diff
@@ -1,66 +1,217 @@
 package ovirt
 
 import (
+	"crypto/tls"
 	"fmt"
-	"net/url"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
 
+	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 	"gopkg.in/AlecAivazis/survey.v1"
 )
 
+// commands definitions
+const (
+	SudoCommand  = "/usr/bin/sudo"
+	CpCommand    = "/usr/bin/cp"
+	TrustCommand = "/usr/bin/trust"
+	CurlCommand  = "/usr/bin/curl"
+	ChmodCommand = "/usr/bin/chmod"
+)
+
+// checkURLResponse performs a GET on the provided urlAddr to ensure that
+// the url actually exists. Users can set skipVerify as true or false to
+// avoid cert validation. In case of failure, returns error.
+func checkURLResponse(urlAddr string, skipVerify bool) error {
+
+	logrus.Debugf("Checking URL Response... urlAddr: %s skipVerify: %s", urlAddr, strconv.FormatBool(skipVerify))
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: skipVerify},
+	}
+
+	client := &http.Client{Transport: tr}
+	resp, err := client.Get(urlAddr)
+	if err != nil {
+		return errors.Wrapf(err, "Error checking URL response")
+	}
+	defer resp.Body.Close()
+
+	return nil
+}
+
+// execCommand executes a command from cmdName with args
+// provided from cmdArgs. Returns the stdout in []byte and error or nil
+func execCommand(cmdName string, cmdArgs []string) ([]byte, error) {
```

Comment:
This looks like a perfect use case for a variadic function:
https://golangbot.com/variadic-functions/

Reply:
I don't think the change is required because inside the function we use Variadic Parameter as Array right? However, I am happy to change if you guys think it is something required. Additionally, we could do it in a different PR.

Reply:
Well you have added it here so I think if that is a new function then we should implement it the way we want.
The benefits of changing it are:
- Little work: inside the function you will use it as an array so no need to change the implementation.
- No need to create a slice before each call.
- Code will be cleaner

Example:
current:
OptionsCurl := []string{}
_, err = execCommand(CurlCommand, OptionsCurl)

variadic:
_, err = execCommand(CurlCommand, "-k", EngineConfig.PemURL, "-o", tmpFile.Name())

Reply:
Alright, let me prepare a patch.

Reply:
I must admit, it's better looking now :) Thanks @Gal-Zaidman